### PR TITLE
Add subscription settings icon dropdown example! 🎉 🎉 🎉

### PR DESCRIPTION
**Pull request by TomerAberbach**
I've been working on a nice subscription settings icon dropdown example for the website!

It's not quite done (hence the draft request), but I thought I'd open the PR to get some feedback. I think it's a bit wordy because I'm trying to accommodate a potentially less knowledgeable audience, but I'd be open to trimming it down with some guidance.

Let me know what you think!

### CODE REVIEW

**Comment by patrickkettner** [comment]
Hi @TomerAberbach!
Thanks so much for submitting - as you can guess, a lot of the team is out for the holidays and as a result this will be slower to be reviewed. 

Greatly appreciate your work, and your patience!

**Comment by TomerAberbach** [comment]
No problem! Take your time 😃

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+author: TomerAberbach
+formats:
+  - email
+  - websites
+--->
+
+<!--
+  ## Introduction
+
+  This example shows how to implement a subscription settings icon dropdown. It is perfect for subscription based emails, as it allows the user to easily change their subscription settings at the moment they realize they want to. It also works for websites.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:13]:
```suggestion
  This example shows how to implement a subscription settings icon dropdown. This lets the user to change their subscription settings without navigating away from the email. It also works for websites.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    <title>Subscription Settings</title>
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+
+    <script async src="https://cdn.ampproject.org/v0.js"></script>
+    <style amp4email-boilerplate>body{visibility:hidden}</style>
+
+    <!-- ## Setup -->
+
+    <!--
+      We need an [`amp-list`](/documentation/components/amp-list) to query the server for the user's current subscription setting on email or page load.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:30]: I'm not sure if this link will work - @matthiasrohmer to confirm please :)

Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:30]:
```suggestion
      We use an [`amp-list`](/documentation/components/amp-list) component to query the user's current subscription setting on email or page load from the server.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+    -->
+    <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
+
+    <!--
+      We'll use [`amp-mustache`](/documentation/components/amp-mustache) to render the icon dropdown in its initial state, which should be based on the server's response.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:35]:
```suggestion
      We'll use [`amp-mustache`](/documentation/components/amp-mustache) o render the dropdown icon linked to the server responded subscription state.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+    -->
+    <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+
+    <!--
+      We'll also need [`amp-form`](/documentation/components/amp-form) to update the user's subscription setting on the server when they select a new setting from the icon dropdown.
+    -->
+    <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
+
+    <!--
+      Lastly, we need [`amp-bind`](/documentation/components/amp-bind) to update the dropdown's state in response to events.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:45]:
```suggestion
      Lastly, we use [`amp-bind`](/documentation/components/amp-bind) to update the dropdown's state in response to events.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+    -->
+    <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
+
+    <!--
+      We'll use [`amp-mustache`](/documentation/components/amp-mustache) to render the icon dropdown in its initial state, which should be based on the server's response.
+    -->
+    <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+
+    <!--
+      We'll also need [`amp-form`](/documentation/components/amp-form) to update the user's subscription setting on the server when they select a new setting from the icon dropdown.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:40]:
```suggestion
      We'll also use [`amp-form`](/documentation/components/amp-form) to update the user's subscription setting on the server when they select a new setting from the icon dropdown.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+          },
+          {
+            "value": "ignoring",
+            "isSelected": false,
+            "text": "Ignoring"
+          }
+        ]
+      }
+      ```
+      `currentSubscription` and `isSelected` vary depending on the user's current subscription setting and `isSelected` is only `true` for one of the objects in `options` at a time.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:139]:
```suggestion
      `currentSubscription` and `isSelected` vary depending on the user's current subscription setting. `isSelected` is only `true` for one of the objects in `options` at a time.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+
+      The server expects POST requests to specify one of the subscription settings above (e.g. `ignoring`) for a `nextSubscription` field in its form data, and will update the user's current subscription setting to the specified value.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:141]:
```suggestion
      The server expects POST requests to specify one of the subscription settings above (e.g. `ignoring`) for a `nextSubscription` field in its form data. It then updates the user's current subscription setting to the specified value.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 8]
```diff
@@ -0,0 +1,434 @@
...
+    <select>
+      <option value="watching">Watching</option>
+      <option value="only-mentions">Only mentions</option>
+      <option value="ignoring">Ignoring</option>
+    </select>
+
+    <!--
+      #### Step 2: Fetching the Current Subscription Setting
+
+      To query the server for the user's current subscription setting on email or page load and pre-select the user's current setting in the dropdown, we wrap the `select` element in an `amp-list` and set its `src` to the URL where our server is listening for GET requests.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:160]:
```suggestion
We wrap the `select` element in an `amp-list` and set its `src` to the URL our server is listening for GET requests. This queries the user's current subscription setting.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 9]
```diff
@@ -0,0 +1,434 @@
...
+    </amp-list>
+
+    <!--
+      Note that the `amp-list` is annotated with [`single-item`](/documentation/components/amp-list#single-item) because our server responds to GET requests with a single logical value (the subscription setting) and we only want the template to be invoked once for the response. Setting [`items="."`](/documentation/components/amp-list#items) ensures that the template can access the response object's root fields.
+    -->
+
+    <!--
+      #### Step 3: Updating the Subscription Setting on Selection
+
+      To update the user's subscription setting on the server when they select a new setting, we start by wrapping the `select` element in an [AMP `form`](/documentation/components/amp-form) element, setting its `method` to `post`, and setting its `action-xhr` attribute to the URL where our server is listening for POST requests.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:194]:
```suggestion
      To update the user's subscription setting on the server when they select a new setting, we start by wrapping the `select` element in an [`amp-form`](/documentation/components/amp-form) element, setting its `method` to `post`, and setting its `action-xhr` attribute to the URL where our server is listening for POST requests.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 10]
```diff
@@ -0,0 +1,434 @@
...
+
+      The `name` attribute of the `select` element is set to `nextSubscription` because `nextSubscription` is the name of the form data field where the server expects the user's new subscription setting to be in the POST request (see the [Server section](<% base_path %>/#server)) and the `select` element is the element containing the user's selected subscription setting.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:196]:
```suggestion
      The `name` attribute of the `select` element is set to `nextSubscription`. `nextSubscription` is the name of the form data field where the server expects the user's new subscription setting to be in the POST request (see the [Server section](<% base_path %>/#server)) and the `select` element contains the user's selected subscription setting.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 11]
```diff
@@ -0,0 +1,434 @@
...
+
+      Finally, we trigger the form submission when the `select` element's value is changed by using the [`on` attribute](/documentation/guides-and-tutorials/learn/common_attributes#on) to attach an event handler to the `select` element's [`change` event](/documentation/guides-and-tutorials/learn/amp-actions-and-events#input-elements) that invokes the `form` element's [`submit` action](/documentation/guides-and-tutorials/learn/amp-actions-and-events#form_1). Note that we gave the `form` element an `id` so that we could refer to it in the event handler.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:198]:
```suggestion
      Finally, we trigger the form submission when the `select` element's value changes. We do so by using the [`on` attribute](/documentation/guides-and-tutorials/learn/common_attributes#on) to attach an event handler to the `select` element's [`change` event](/documentation/guides-and-tutorials/learn/amp-actions-and-events#input-elements) , invoking the `form` element's [`submit` action]. [`submit` action](/documentation/guides-and-tutorials/learn/amp-actions-and-events#form_1). Note that we gave the `form` element an `id` so that we could refer to it in the event handler.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 12]
```diff
@@ -0,0 +1,434 @@
...
+
+      You can test that the code is functional by selecting a new subscription setting, refreshing the page, and noticing that the `select` element's subscription setting on page load is the one you selected!
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:200]:
```suggestion
      You can test the code by selecting a new subscription setting and refreshing the page. The `select` element's subscription setting on page load is now the one you selected!
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 13]
```diff
@@ -0,0 +1,434 @@
...
+
+      We start by adding a hidden `input` element, and moving the `name` of the `select` element to the hidden `input` element so that its value is submitted. To keep the hidden `input` element's value in sync with the `select` element's value we update a new `nextSubscription` state variable using [`AMP.setState`](/documentation/components/amp-bind#updating-state-variables-with-amp.setstate()) whenever the `select` element's value changes and [bind](/documentation/components/amp-bind#bindings) the hidden `input` element's `value` attribute to the state variable.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:232]:
```suggestion
      We start by adding a hidden `input` element, and moving the `name` of the `select` element to the hidden `input` element so that its value submits. To keep the hidden `input` element's value in sync with the `select` element's value, we update a new `nextSubscription` state variable using [`AMP.setState`](/documentation/components/amp-bind#updating-state-variables-with-amp.setstate()) whenever the `select` element's value changes and [bind](/documentation/components/amp-bind#bindings) the hidden `input` element's `value` attribute to the state variable.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 14]
```diff
@@ -0,0 +1,434 @@
...
+    </amp-list>
+
+    <!--
+      Note that despite the `disabled` attribute being a [boolean attribute](https://www.w3.org/TR/2011/WD-html5-author-20110809/common-microsyntaxes.html#boolean-attribute), we can still effectively bind it because [`amp-bind` will remove the attribute when the binding expression is `false`](/documentation/components/amp-bind#bindings:~:text=Boolean%20expression%20results%20toggle%20boolean%20attributes.,false%2C%20the%20controls%20attribute%20is%20removed.).
+    -->
+
+    <!--
+      #### Step 5: Handling Form Submission Failure
+
+      When the form fails to submit we should display an error message and revert the `select` element to its previous value so that the user interface is consistent with the user's persisted subscription setting.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:271]:
```suggestion
      We need to display an error message and revert the `select` element to its previous value if the form fails to submit.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 15]
```diff
@@ -0,0 +1,434 @@
...
+
+      To display an error message when the form fails to submit we add element containing our error message to the `form` element's children, and annotate the child element with the [`submit-error` attribute](/documentation/components/amp-form#submit-error). The attribute ensures that the element is only visible after a failed form submission.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:273]:
```suggestion
      To display an error message when the form fails to submit, we add element containing our error message to the `form` element's children. We annotate the child element with the [`submit-error` attribute](/documentation/components/amp-form#submit-error). This attribute ensures that the element is only visible after a failed form submission.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 16]
```diff
@@ -0,0 +1,434 @@
...
+
+      Reverting the `select` element's value to its previous value requires fully managing the `select` element's value by binding the `selected` attribute of each of its `option` elements. This requires maintaining the `select` element's current value in a `currentSubscription` state variable and the element's previous value in a `previousSubscription` state variable.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:275]:
```suggestion
      Reverting the `select` element's value to its previous value requires binding the `selected` attribute of each `option` element. This requires maintaining the `select` element's current value in a `currentSubscription` state variable and the element's previous value in a `previousSubscription` state variable.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 17]
```diff
@@ -0,0 +1,434 @@
...
+
+      When the `select` element's value is changed we set `currentSubscription` to the new value and `previousSubscription` to the element's previous value, which is stored in the `currentSubscription` state variable if this is not the first time the element's value has been changed or in the `currentSubscription` field of the `amp-list` response otherwise.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:277]:
```suggestion
      When the `select` element's value changes, we set `currentSubscription` to the new value and `previousSubscription` to the element's previous value. We store these in the `currentSubscription` state variable, if this is not the first time the element's value has changed. Otherwise, it's stored in the `currentSubscription` field of the `amp-list` response.
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 18]
```diff
@@ -0,0 +1,434 @@
...
+        </form>
+      </template>
+      <div placeholder>Loading...</div>
+      <div fallback>Something went wrong. Please refresh</div>
+    </amp-list>
+
+    <!--
+      #### Step 6: Icon Dropdown
+
+      Styling the `select` element itself to look like an icon using only CSS supported by email clients isn't feasible. Instead, we'll hide the `select` element's persistent user interface, which shows the element's current value, and show a different icon for each dropdown state in its place, but keep the element's ephemeral dropdown, what you see on click.
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:327]: The second sentence in this section is a little long and confusing. Could you try breaking it up for clarity?

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 19]
```diff
@@ -0,0 +1,434 @@
...
+
+      Fortunately, styling `select` primarily affects the persistent user interface while styling `option` primarily affects the ephemeral dropdown. We can use this fact to hide the former, but keep the latter. Of the three common choices for hiding an element using CSS ([`display: none;`, `visibility: hidden;`, and `opacity: 0;`](https://magnusbenoni.com/difference-between-display-visibility-opacity)), `opacity: 0;` is most suited because we want to keep the element in the tab order as well as keep receiving its click events:
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:329]:
```suggestion
      Fortunately, styling `select` affects the persistent user interface while styling `option` affects the ephemeral dropdown. We can use this fact to hide the former, but keep the latter. Of the three common choices for hiding an element using CSS ([`display: none;`, `visibility: hidden;`, and `opacity: 0;`](https://magnusbenoni.com/difference-between-display-visibility-opacity)), `opacity: 0;` is most suited because we want to keep the element in the tab order as well as keep receiving its click events:
```

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html [hunk 20]
```diff
@@ -0,0 +1,434 @@
...
+      .subscription-select {
+        opacity: 0;
+        cursor: pointer;
+      }
+      ```
+      We also set `cursor: pointer;` for the `select` element because it looks more like a button now.
+
+      The next step is to create a new icon-based user interface and position it _behind_ the invisible `select` user interface (in terms of stacking order) so that when a user tries to click on one of our icons they'll actually be clicking on the invisible `select` element and triggering its events! We'll keep our new icons behind the `select` element by placing the icon's code before the `select` element's code (see [Stacking without the `z-index` property](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/Stacking_without_z-index)).
+
+      The `relative` and `absolute` CSS classes are used to position the `select` element above the icons, and the `icon-size` class is used to ensure the element's click target bounding box is the same size as the icons:
```
Comment (CrystalOnScript) [inline examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:340]:
```suggestion
      We use the `relative` and `absolute` CSS classes are to position the `select` element above the icons, and use the `icon-size` class to ensure the element's click target bounding box is the same size as the icons:
```

**Review by CrystalOnScript — COMMENTED** [review]
Overall this looks really awesome! I've left a few suggestions that clean up passive voice and break up a few sentences for clarity. There was one I didn't quite understand, so please take a look :) 

Thanks so much for your help!

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html (hunk 9 above)
Comment (TomerAberbach) [reply examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:194]: I wasn't sure about this one because the usage in the code is still `form` (not `amp-form`). Do you still think we should change it to `amp-form`?

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html (hunk 18 above)
Comment (TomerAberbach) [reply examples/source/interactivity-dynamic-content/Subscription_Settings/index.html]: Done!

**Comment by TomerAberbach** [comment]
I think I'm ready for a final review! Although, I wasn't able to test this most recent commit on my local server. For some reason I'm getting the following error when running `npm run develop -- --queue-imports`:
```
[12:06:23 AM] › ✔  success   Configured Grow!
[12:06:23 AM] [Grow] › ℹ  info      Will extract page infos for highlight list /home/tomer/Documents/amp.dev/platform/config/search-promoted-pages.json
[12:06:24 AM] [Grow] › ℹ  info      Page infos for highlight list /home/tomer/Documents/amp.dev/platform/config/search-promoted-pages.json written for all locales into /home/tomer/Documents/amp.dev/dist/static/files/search-promoted-pages
[12:06:26 AM] [Grow] › ℹ  info      1391 routes built in 1.982 s
[12:06:26 AM] [Grow] › ✔  success   Grow started successfully.
[12:06:26 AM] [Page Cache] › ⚠  warning   No Redis instances available. Falling back to LRU
[00:06:26] 'run' errored after 5.62 s
[00:06:26] Error: ENOENT: no such file or directory, open '/home/tomer/Documents/amp.dev/dist/pages/404.html'
    at Object.openSync (node:fs:488:3)
    at readFileSync (node:fs:389:35)
    at Object.<anonymous> (/home/tomer/Documents/amp.dev/platform/lib/routers/notFound.js:26:20)
    at Module._compile (node:internal/modules/cjs/loader:1102:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1131:10)
    at Module.load (node:internal/modules/cjs/loader:967:32)
    at Function.Module._load (node:internal/modules/cjs/loader:807:14)
    at Module.require (node:internal/modules/cjs/loader:991:19)
    at require (node:internal/modules/cjs/helpers:92:18)
    at Object.<anonymous> (/home/tomer/Documents/amp.dev/platform/lib/platform.js:46:13)
    at Module._compile (node:internal/modules/cjs/loader:1102:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1131:10)
    at Module.load (node:internal/modules/cjs/loader:967:32)
    at Function.Module._load (node:internal/modules/cjs/loader:807:14)
    at Module.require (node:internal/modules/cjs/loader:991:19)
    at require (node:internal/modules/cjs/helpers:92:18)

```

I tried cloning and setting up my repository again, but it didn't seem to do anything.

File: examples/source/interactivity-dynamic-content/Subscription_Settings/index.html (hunk 9 above)
Comment (CrystalOnScript) [reply examples/source/interactivity-dynamic-content/Subscription_Settings/index.html:194]: hmm, valid point! I think the distinction here is we're saying "element" instead of "component" - it is called an `amp-form` component, but the element is just `form`. In this case, you can leave as AMP `form`.

**Review by CrystalOnScript — APPROVED** [review]
